framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1310,10 +1310,9 @@ NonlinearSystemBase::computeResidualInternal(const std::set<TagID> & tags)
 
     Threads::parallel_reduce(elem_range, cr);
 
-    unsigned int n_threads = libMesh::n_threads();
-    for (unsigned int i = 0; i < n_threads;
-         i++) // Add any cached residuals that might be hanging around
-      _fe_problem.addCachedResidual(i);
+    // Add any cached residuals that might be hanging around
```

Comment:
Similar

Reply:
This is the API: `NonlinearSystemBase::computeResidualInternal(const std::set<TagID> & tags)`

Reply:
You really should use `computeResidualTags` and do your own `associateVectorToTag` and `disassociateVectorFromTag`, and you can control the things whatever way you want.